modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -565,6 +565,60 @@ const Fields = ({
 		}
 	};
 
+	const handleDelete = ({item}: {item: FDSField}) => {
+		openModal({
+			bodyHTML: Liferay.Language.get(
+				'are-you-sure-you-want-to-delete-this-field?-fragments-using-it-will-be-affected'
+			),
+			buttons: [
+				{
+					autoFocus: true,
+					displayType: 'secondary',
+					label: Liferay.Language.get('cancel'),
+					type: 'cancel',
+				},
+				{
+					displayType: 'warning',
+					label: Liferay.Language.get('delete'),
+					onClick: ({processClose}: {processClose: Function}) => {
+						processClose();
+
+						const url = `${API_URL.FDS_FIELDS}/${item.id}`;
+
+						fetch(url, {
+							method: 'DELETE',
+						})
+							.then(() => {
+								openToast({
+									message: Liferay.Language.get(
+										'your-request-completed-successfully'
+									),
+									type: 'success',
+								});
+
+								setFDSFields(
+									fdsFields?.filter(
+										(fdsField: FDSField) =>
+											fdsField.id !== item.id
+									) || []
+								);
+							})
+							.catch(() => {
+								openToast({
+									message: Liferay.Language.get(
+										'your-request-failed-to-complete'
+									),
+									type: 'danger',
+								});
+							});
```

Comment:
Would this accept an async/await here instead of using a promise?

Reply:
Yup, seems to work. I'll switch to using async/await.